### PR TITLE
[Spree 2.1] Fix issues in EnterprisesController

### PR DESCRIPTION
**Pull request by luisramos0**
#### What? Why?

Fixes two problems in Api enterprise controller. See commit message.

Fixes spec/features/consumer/registration_spec.rb:24

#### What should we test?

Green spec.

### CODE REVIEW

File: app/controllers/api/enterprises_controller.rb [hunk 1]
```diff
@@ -72,5 +76,15 @@ def override_sells
...
     end
+
+    def save_enterprise_users(user_ids)
+      return unless user_ids
+
+      user_ids.each do |user_id|
+        next if @enterprise.user_ids.include? user_id.to_i
+
+        @enterprise.users << Spree::User.find(user_id)
+      end
```
Comment (sauloperez) [inline app/controllers/api/enterprises_controller.rb]: I believe we can let ActiveRecord do all this for us (and possibly a bit more efficiently). Looking at the associations Rails guide [collection_singular_ids=(ids)](https://guides.rubyonrails.org/association_basics.html#methods-added-by-has-many-collection-singular-ids-ids) seems to be what we want.

Comment (luisramos0) [reply app/controllers/api/enterprises_controller.rb]: I started by trying
@enterprise.user_ids << user_ids

that's what collecction_singular_ids means, right?
but it doesnt work!

Comment (sauloperez) [reply app/controllers/api/enterprises_controller.rb]: I don't know if you already did it but this method is a setter so should be used like

```ruby
@enterprise.user_ids = user_ids
```

Comment (luisramos0) [reply app/controllers/api/enterprises_controller.rb]: ah, no, maybe that's it :face_with_head_bandage: :see_no_evil:

Comment (luisramos0) [reply app/controllers/api/enterprises_controller.rb]: nice, it works. see new commit.
thanks for the lesson!!!

**Pull request merged**